internal/charmstore/search_test.go

```diff
@@ -130,7 +133,37 @@ func (s *StoreSearchSuite) TestExportOnlyLatest(c *gc.C) {
 	c.Assert(err, gc.IsNil)
 	err = s.store.ES.GetDocument(s.TestIndex, typeName, s.store.ES.getID(old.URL), &actual)
 	c.Assert(err, gc.IsNil)
-	doc := SearchDoc{Entity: expected, ReadACLs: []string{"charmers", params.Everyone}}
+	doc := SearchDoc{
+		Entity:   expected,
+		ReadACLs: []string{"charmers", params.Everyone},
+		Series:   expected.SupportedSeries,
+	}
+	c.Assert(string(actual), jc.JSONEquals, doc)
+}
+
+func (s *StoreSearchSuite) TestExportMultiSeriesCharmsReplaceEarlierOnes(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmDir("wordpress")
+	url := newResolvedURL("cs:~charmers/trusty/juju-gui-24", -1)
+	err := s.store.AddCharmWithArchive(url, charmArchive)
+	c.Assert(err, gc.IsNil)
+	charmArchive = storetesting.Charms.CharmDir("juju-gui")
+	url = newResolvedURL("cs:~charmers/juju-gui-25", -1)
+	err = s.store.AddCharmWithArchive(url, charmArchive)
```

Comment:
To the question - no because the juju-gui charm does have supported series in the metadata, the wordpress one does not, but that has a series in the URL.
To the second point, I'll add a test.